File: app/src/main/java/com/chessclub/app/ui/rankings/PlayerAdapter.java
package com.chessclub.app.ui.rankings;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.chessclub.app.R;
import com.chessclub.app.model.Player;

import java.util.List;

/**
 * Adapter for displaying player items in a RecyclerView
 */
public class PlayerAdapter extends RecyclerView.Adapter<PlayerAdapter.PlayerViewHolder> {
    private List<Player> players;

    /**
     * Constructor
     * @param players List of players to display
     */
    public PlayerAdapter(List<Player> players) {
        this.players = players;
    }

    @NonNull
    @Override
    public PlayerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_player, parent, false);
        return new PlayerViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PlayerViewHolder holder, int position) {
        Player player = players.get(position);
        
        // Set position number (ranking)
        holder.tvRank.setText(String.valueOf(position + 1));
        
        // Set player name
        holder.tvName.setText(player.getName());
        
        // Set player ELO rating
        holder.tvElo.setText(String.valueOf(player.getElo()));
        
        // Set player stats
        String stats = player.getWins() + "W / " + player.getDraws() + "D / " + player.getLosses() + "L";
        holder.tvStats.setText(stats);
        
        // Calculate and set win rate
        float winRate = player.getWinRate();
        holder.tvWinRate.setText(String.format("%.1f%%", winRate));
        
        // Set games played
        holder.tvGamesPlayed.setText(String.valueOf(player.getGamesPlayed()));
    }

    @Override
    public int getItemCount() {
        return players == null ? 0 : players.size();
    }

    /**
     * Update the adapter with new players data
     * @param newPlayers New list of players
     */
    public void updatePlayers(List<Player> newPlayers) {
        this.players = newPlayers;
        notifyDataSetChanged();
    }

    /**
     * ViewHolder for player items
     */
    static class PlayerViewHolder extends RecyclerView.ViewHolder {
        TextView tvRank;
        TextView tvName;
        TextView tvElo;
        TextView tvStats;
        TextView tvWinRate;
        TextView tvGamesPlayed;

        public PlayerViewHolder(@NonNull View itemView) {
            super(itemView);
            tvRank = itemView.findViewById(R.id.tv_rank);
            tvName = itemView.findViewById(R.id.tv_name);
            tvElo = itemView.findViewById(R.id.tv_elo);
            tvStats = itemView.findViewById(R.id.tv_stats);
            tvWinRate = itemView.findViewById(R.id.tv_win_rate);
            tvGamesPlayed = itemView.findViewById(R.id.tv_games_played);
        }
    }
}

File: ChessClubAndroid/app/src/main/java/com/chessclub/android/ui/rankings/PlayerAdapter.kt
package com.chessclub.android.ui.rankings

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.chessclub.android.R
import com.chessclub.android.model.Player

class PlayerAdapter : ListAdapter<Player, PlayerAdapter.PlayerViewHolder>(PlayerDiffCallback()) {
    
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlayerViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_player, parent, false)
        return PlayerViewHolder(view)
    }
    
    override fun onBindViewHolder(holder: PlayerViewHolder, position: Int) {
        val player = getItem(position)
        holder.bind(player, position + 1) // position + 1 for rank
    }
    
    class PlayerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val rankTextView: TextView = itemView.findViewById(R.id.rank)
        private val nameTextView: TextView = itemView.findViewById(R.id.player_name)
        private val ratingTextView: TextView = itemView.findViewById(R.id.rating)
        private val gamesPlayedTextView: TextView = itemView.findViewById(R.id.games_played)
        
        fun bind(player: Player, rank: Int) {
            rankTextView.text = rank.toString()
            nameTextView.text = player.name
            ratingTextView.text = player.rating.toString()
            gamesPlayedTextView.text = player.gamesPlayed.toString()
            
            // Apply special styling for top 3 players
            if (rank <= 3) {
                itemView.setBackgroundResource(R.drawable.top_player_background)
                nameTextView.setTextAppearance(R.style.TextAppearance_AppCompat_Medium)
                nameTextView.setTypeface(null, android.graphics.Typeface.BOLD)
            } else {
                itemView.setBackgroundResource(0)
                nameTextView.setTextAppearance(R.style.TextAppearance_AppCompat_Body1)
                nameTextView.setTypeface(null, android.graphics.Typeface.NORMAL)
            }
        }
    }
    
    class PlayerDiffCallback : DiffUtil.ItemCallback<Player>() {
        override fun areItemsTheSame(oldItem: Player, newItem: Player): Boolean {
            return oldItem.id == newItem.id
        }
        
        override fun areContentsTheSame(oldItem: Player, newItem: Player): Boolean {
            return oldItem == newItem
        }
    }
}
